### PR TITLE
Cpp11 initializers 2

### PR DESCRIPTION
**Pull request by nerzhul**
continue previous work: cleaning up constructors. Also fix some NULL checks on nullptr replaced values to be coherent & portable between NULL & nullptr

### CODE REVIEW

File: src/guiChatConsole.cpp [hunk 1]
```diff
@@ -76,12 +76,9 @@ GUIChatConsole::GUIChatConsole(
...
 	m_font = g_fontengine->getFont(FONT_SIZE_UNSPECIFIED, FM_Mono);
 
-	if (m_font == NULL)
-	{
+	if (!m_font)
 		errorstream << "GUIChatConsole: Unable to load mono font ";
-	}
-	else
-	{
+	else {
```
Comment (SmallJoker) [inline src/guiChatConsole.cpp]:
```C++
if (!m_font) {
	
} else {

}
```
as soon one part isn't an one-liner.

File: src/httpfetch.h [hunk 2]
```diff
@@ -69,23 +69,18 @@ struct HTTPFetchRequest
...
-	    : succeeded(false), timeout(false), response_code(0), data(""),
-	      caller(HTTPFETCH_DISCARD), request_id(0)
-	{
-	}
+	HTTPFetchResult() {}
 
 	HTTPFetchResult(const HTTPFetchRequest &fetch_request)
-	    : succeeded(false), timeout(false), response_code(0), data(""),
-	      caller(fetch_request.caller), request_id(fetch_request.request_id)
+		: caller(fetch_request.caller), request_id(fetch_request.request_id)
```
Comment (SmallJoker) [inline src/httpfetch.h]: LINT doesn't like this line

File: src/localplayer.h [hunk 3]
```diff
@@ -143,35 +146,36 @@ class LocalPlayer : public Player
...
-	GenericCAO *m_cao;
+	v3s16 m_old_node_below = v3s16(32767, 32767, 32767);
+	std::string m_old_node_below_type = "air";
+	bool m_can_jump = false;
+	u16 m_breath = PLAYER_MAX_BREATH;
+	f32 m_yaw = 0.0f;
+	f32 m_pitch = 0.0f;
+	bool camera_barely_in_ceiling = false;
+	aabb3f m_collisionbox = aabb3f(-BS * 0.30f, 0.0f, -BS * 0.30f, BS * 0.30f,
+			BS * 1.75f, BS * 0.30f);
```
Comment (SmallJoker) [inline src/localplayer.h]: Could you please add the new line an argument earlier, so the X,Y,Z values are grouped nicely

**Comment by SmallJoker** [comment]
Apart from the comments above, LGTM.

**Comment by nerzhul** [comment]
ty for review i will fix them soon

**Comment by nerzhul** [comment]
@SmallJoker all points fixed

**Pull request merged**